loky/process_executor.py

```diff
@@ -1269,8 +1282,11 @@ def shutdown(self, wait=True, kill_workers=False):
                 self._executor_manager_thread_wakeup.wakeup()
 
         if executor_manager_thread is not None and wait:
-            executor_manager_thread.join()
-            _threads_wakeups.pop(executor_manager_thread, None)
+            # This locks avoids concurrent join if the interpreter
+            # is shutting down.
```

Comment:
Do you understand why a concurrent call to join could call a deadlock? joining a terminated thread should not block, or am I missing something?

Reply:
This does not create a deadlock, just weird errors with pypy when calling join.